zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/DynamicCredentialsFileLoader.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.server.internal.elasticsearch;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Properties;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.PASSWORD_PROP;
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.USERNAME_PROP;
+
+/**
+ * Load username/password from credentials file.
+ */
+class DynamicCredentialsFileLoader {
+  static final String CREDENTIALS_FILE_REFRESH_INTERVAL_IN_SECOND =
+    "zipkin.storage.elasticsearch.credentials-file-refresh-interval-in-second";
+
+  private final String credentialsFilePath;
+
+  private final BasicCredentials basicCredentials;
+
+  public DynamicCredentialsFileLoader(BasicCredentials basicCredentials,
+    String credentialsFilePath) {
+    this.basicCredentials = basicCredentials;
+    this.credentialsFilePath = credentialsFilePath;
+  }
+
+  @Scheduled(fixedRateString = "${" + CREDENTIALS_FILE_REFRESH_INTERVAL_IN_SECOND + "}")
+  void load() throws IOException {
```

Comment:
throwing the `IOException` can get quite spammy if the refresh interval is small. Should we consider logging the stacktrace only once, and then suppressing it for further invocations ?

Reply:
agree let's not throw and instead log. log once is ok or rate-limited if want to use something like armeria's rate limited logging (not sure it is a generally available api)

Reply:
I opt to guava RateLimiter to log an exception every 10 seconds.